server/camphoric/models.py

```diff
@@ -220,6 +223,19 @@ class Registration(TimeStampedModel):
     def __str__(self):
         return "Registration #{} ({})".format(self.id, self.event.name)
 
+    def recalculate_server_pricing(self):
+        server_pricing_results = pricing.calculate_price(
+            self,
+            self.campers.all()
+        )
+        self.server_pricing_results = server_pricing_results
+        super().save()
+
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)  # Call the "real" save() method.
+        # calculate price on change
+        self.recalculate_server_pricing()
```

Comment:
It looks like each time a camper is saved (looking also at Camper.save below), the price will be recalculated and the registration will be saved twice, which would be a lot of redundant work. Possible alternative:

- in the overridden Registration.save(), take an optional `campers` kwarg
- if supplied, pass `campers` to calculate_price instead of self.campers.all()
- the only time you'd need to pass this would be in post_registration, where the registration doesn't yet have campers in the DB

so this save() might look like

```
def save(self, *args, **kwargs):
     campers = del kwargs['campers'] if 'campers' in kwargs else self.campers.all()
     self.server_pricing_results = pricing.calculate_price(self, campers)
     super().save(*args, **kwargs)
```

I would suggest not overriding Campers.save(), but saving the registration (after saving the campers) each time you want to update the pricing results.

Reply:
Your updated code makes sense.  re: the campers.save override - part of the reason I did that is because if a camper admin updates _just_ a camper attribute (which could affect pricing) I also want a price recalc to be triggered.